src/map.js

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import L from "leaflet";
+import "leaflet/dist/leaflet.css";
+import icon from "leaflet/dist/images/marker-icon.png";
+import iconShadow from "leaflet/dist/images/marker-shadow.png";
+
+export default function MapComponent({ searchedLocation }) {
+  const [map, setMap] = useState(null);
+
+  useEffect(() => {
+    // create a new leaflet map instance
+    const leafletMap = L.map("map");
+
+    // add a tile layer from OpenStreetMap
+    L.tileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
+      attribution:
+        'Map data © <a href="https://openstreetmap.org">OpenStreetMap</a> contributors',
+      maxZoom: 19,
+    }).addTo(leafletMap);
+
+    // define the default marker icon
+    const DefaultIcon = L.icon({
+      iconUrl: icon,
+      shadowUrl: iconShadow,
+      iconSize: [25, 41],
+      iconAnchor: [12, 41],
+    });
+
+    // set the default marker icon for all markers on the map
+    L.Marker.prototype.options.icon = DefaultIcon;
+
+    // set the map instance to the state
+    setMap(leafletMap);
+
+    return () => {
+      // clean up the map instance when the component unmounts
+      leafletMap.remove();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (map && searchedLocation) {
+      // update the map view to the searched location
+      map.setView(searchedLocation, 13);
+
+      // add a marker to the map at the searched location
+      const marker = L.marker(searchedLocation).addTo(map);
+
+      // add a popup to the marker
+      marker.bindPopup("Searched Location").openPopup();
+    }
+  }, [map, searchedLocation]);
+
+  return (
+    <div id="map" style={{ height: "400px" }}>
+      {!map && <p>Loading map...</p>}
+    </div>
+  );
+}
```

Comment:
Somewhere, there's a bug (or maybe a feature), please investigate (or do an RCA) - 

When I just search for New York City, there's one marker - 

![Screenshot 2023-03-16 at 23 46 19](https://user-images.githubusercontent.com/19688420/225776715-49f15d1d-ae58-460e-9bc4-c0a67da7e098.png)


When I search for London, it points to London, UK, but also adds a new marker below - Where and Why is it coming? (I have just searched for two cities, but there are 3 markers on the map)

![Screenshot 2023-03-16 at 23 47 53](https://user-images.githubusercontent.com/19688420/225776884-e0a7e653-2041-40c2-aa39-7e0bcd32556c.png)

Reply:
I worked on the solution You can check it